File: Front/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _http:HttpClient) { }

  public quizess(){
    return this._http.get(`${baseUrl}quiz/`)
  }
  public addQuiz(quiz:any){
    return this._http.post(`${baseUrl}quiz/`,quiz);
  }
  // delete quiz
  public deleteQuiz(quizId: any){
    return this._http.delete(`${baseUrl}quiz/${quizId}`);
  }

  // update quiz

  public updateQuiz(quiz:any){
    return this._http.put(`${baseUrl}quiz/`,quiz);
  }

  // get quiz by id
  public getQuizBy(quizId: any){
    return this._http.get(`${baseUrl}quiz/${quizId}`);
  }

  // get quizzes of category
  public getQuizzesOfCategory(catId: any){
    return this._http.get(`${baseUrl}quiz/category/${catId}`);
  }

}

File: examserver/src/main/java/com/exam/examserver/service/QuizService.java
package com.exam.examserver.service;

import com.exam.examserver.model.Exam.Category;
import com.exam.examserver.model.Exam.Quiz;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Set;

public interface QuizService {
    public Quiz addQuiz(Quiz quiz);
    public  Quiz updateQuiz(Quiz quiz);
    public Set<Quiz> getQuizzes();
    public Quiz getQuiz(Long quizId);
    public void deleteQuiz(Long quizId);


    public List<Quiz> getQuizzesOfCategory(Category category);
    public List<Quiz> getActiveQuizzes();
    public List<Quiz>getActiveQuizzesOfCategory(Category category);
}
